### PR TITLE
Support for Globus SDK 3

### PR DESCRIPTION
**Pull request by WardLT**
Will break support for previous versions of Globus SDK, so I edited the requirements as appropriate.

### CODE REVIEW

**Comment by ascourtas** [comment]
Going to try closing and reopening to trigger tests

**Comment by ascourtas** [comment]
attempt to retrigger tests

**Comment by ascourtas** [comment]
Currently in the process of bringing the DLHub-specific funcx helm chart up-to-date, and to make the pods that run the DLHub servables inline with Globus SDK v3 -- pause on merging until that is resolved

**Comment by ascourtas** [comment]
DLHub-specific helm chart is updated and DLHub endpoint is redeployed -- now ok to merge once tests pass

File: dlhub_sdk/client.py [hunk 1]
```diff
@@ -111,14 +113,14 @@ def __init__(self, dlh_authorizer: Optional[GlobusAuthorizer] = None,
                                       no_local_server=kwargs.get("no_local_server", True),
                                       no_browser=kwargs.get("no_browser", True))
         self._search_client = globus_sdk.SearchClient(authorizer=search_authorizer,
-                                                      http_timeout=5 * 60)
+                                                      transport_params={"http_timeout": 5 * 60})
```
Comment (WardLT) [inline dlhub_sdk/client.py:116]: Why not use the user-provided timeout here? FWIW, I don't think there's anything anything special about the 5-minute timeout here.

**Review by WardLT — COMMENTED** [review]
One minor change, then I think it's ready to merge 💯

File: dlhub_sdk/client.py (hunk 1 above)
Comment (ascourtas) [reply dlhub_sdk/client.py:116]: oh gotcha! Happy to change it then, I wasn't sure :)

**Comment by ascourtas** [comment]
confirmed with local testing that model publication and running works with this updated branch, AND running models published with previous DLHub SDK (<=0.10.0) works properly

**Pull request merged**